code/modules/bitrunning/objects/watcher_hunt.dm

```diff
@@ -0,0 +1,471 @@
+	var/shot = FALSE
+	var/score = 500
+	var/boulder_size = BOULDER_SIZE_SMALL
+	var/list/boulder_mats = list(
+		/datum/material/iron = SHEET_MATERIAL_AMOUNT * 5,
+		/datum/material/glass = SHEET_MATERIAL_AMOUNT * 5,
+	)
+	var/list/drop_sounds = list('sound/machines/coindrop.ogg', 'sound/machines/coindrop2.ogg')
+	var/starting_location
+	var/obj/bitrunning/watcher_hunt_spawner/my_spawner
+	var/use_human_appearance = FALSE
+	var/human_outfit = /datum/outfit/job/assistant
+	var/human_species = /datum/species/human
+	var/r_hand = null
+	var/l_hand = null
+	var/bloody_slots_outfit = null
+
+
+/obj/bitrunning/target/Initialize(mapload)
+	. = ..()
+	if(use_human_appearance)
+		apply_dynamic_human_appearance(src, human_outfit, human_species, r_hand = r_hand, l_hand = l_hand, bloody_slots = bloody_slots_outfit)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, PROC_REF(check_teleport))
+
+/obj/bitrunning/target/Destroy(force)
+	UnregisterSignal(src, COMSIG_MOVABLE_MOVED)
+	my_spawner.current_targets.Remove(src)
+	. = ..()
+
+/obj/bitrunning/target/bullet_act(obj/projectile/hitting_projectile, def_zone, piercing_hit)
+	. = ..()
+	if(shot)
+		return
+	shot = TRUE
+	my_score.current_score += score * my_score.current_multiplier
+	my_score.update_icon()
+	my_score.check_score()
+	if(score > 0)
+		var/turf/my_turf = get_turf(src)
+		my_turf.balloon_alert_to_viewers("+[score * my_score.current_multiplier]")
+		my_score.current_multiplier++
+		var/obj/item/boulder/reward_boulder = new(get_turf(my_score))
+		reward_boulder.set_custom_materials(boulder_mats)
+
+		//set size & durability
+		reward_boulder.boulder_size = boulder_size
+		reward_boulder.durability = rand(2, boulder_size) //randomize durability a bit for some flavor.
+		reward_boulder.boulder_string = pick(list("boulder", "rock", "stone"))
+		reward_boulder.update_appearance(UPDATE_ICON_STATE)
+
+		SSore_generation.available_boulders += reward_boulder
+	else if (score < 0)
+		var/turf/my_turf = get_turf(src)
+		my_turf.balloon_alert_to_viewers("-[score]!")
+		if(my_score.current_score < 0)
+			my_score.current_score = 0
+		my_score.current_multiplier = 1
+	else
+		my_score.current_multiplier = 1
+	playsound(my_score, pick(drop_sounds), 100, TRUE)
+	qdel(src)
+
+/obj/bitrunning/target/proc/check_teleport(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+	var/obj/bitrunning/watcher_hunt_teleporter/teleport_node = locate(/obj/bitrunning/watcher_hunt_teleporter) in get_turf(src)
+	if(teleport_node)
+		forceMove(starting_location)
+
+/obj/bitrunning/watcher_hunt_spawner
+	name = "Watcher Hunt Spawner"
+	desc = "If you can see this, file a bug report!"
+	icon = 'icons/obj/machines/bitrunning.dmi'
+	base_icon_state = "animatronic_node"
+	icon_state = "animatronic_node"
+	color = "#FF0000"
+	mouse_opacity = 0
+	density = FALSE
+	anchored = TRUE
+	flags_1 = INDESTRUCTIBLE
+	var/list/spawn_order = list()
+	var/list/fucking_real_spawn_order_fuck_you_byond_stupid_fucking_language = list()
```

Comment:
ffs, give it a shorter name. If you have to somehow vent out your internal turmoil, do it in a comment, preferably as documentation to help us understand it.